contracts/core/Dispatcher.sol

```diff
@@ -86,11 +87,20 @@ contract Dispatcher is OwnableUpgradeable, UUPSUpgradeable, IDispatcher {
         L1Header calldata l1header,
         OpL2StateProof calldata proof,
         uint256 height,
-        uint256 appHash
+        uint256 appHash,
+        uint256 clientId
     ) external returns (uint256 fraudProofEndTime, bool ended) {
+        LightClient lightClient = _clientIdToLightClient[clientId];
+        if (address(lightClient) == address(0)) {
+            revert IBCErrors.lightClientNotFound(clientId);
+        }
         return lightClient.addOpConsensusState(l1header, proof, height, appHash);
     }
 
+    function addNewConnection(string calldata connection, LightClient lightClient) external onlyOwner {
```

Comment:
pls document the method params, esp connectionID as the first in connectionHops during channel handshake

Reply:
I added natspec docs in a seperate PR on [this branch](https://github.com/open-ibc/vibc-core-smart-contracts/blob/a9081b6d00c609f9409988c62384a2c16ce596e0/contracts/core/Dispatcher.sol#L842-L844), so I will implement your suggestion in that one if that's ok!

Reply:
added the specific comment about connectionHops [here](https://github.com/open-ibc/vibc-core-smart-contracts/blob/e732841a3900ed6c21b99b6ebbd4e5f135c5dc8d/contracts/core/Dispatcher.sol#L123-L124)